automation/taskcluster/lib/tasks.py

```diff
@@ -2,143 +2,517 @@
+                ),
+            ]
+        return routes
+
+    def craft_detekt_task(self):
+        return self._craft_gradle_clean_task(
+            name='detekt',
+            description='Running detekt over all modules',
+            gradle_task='detekt',
+            treeherder={
+                'jobKind': 'test',
+                'machine': {
+                  'platform': 'lint',
+                },
+                'symbol': 'detekt',
+                'tier': 1,
             }
-        }
 
-    def craft_signing_task(self, build_task_id, name, description, signing_format, is_staging, apks, scopes, routes):
-        created = datetime.datetime.now()
-        expires = taskcluster.fromNow('1 year')
-        deadline = taskcluster.fromNow('1 day')
+        )
 
-        return {
-            "workerType": 'mobile-signing-dep-v1' if is_staging else 'mobile-signing-v1',
-            "taskGroupId": self.task_id,
-            "schedulerId": self.scheduler_id,
-            "expires": taskcluster.stringDate(expires),
-            "retries": 5,
-            "created": taskcluster.stringDate(created),
-            "tags": {},
-            "priority": "lowest",
-            "deadline": taskcluster.stringDate(deadline),
-            "dependencies": [self.task_id, build_task_id],
-            "routes": routes,
-            "scopes": scopes,
-            "requires": "all-completed",
-            "payload": {
-                "maxRunTime": 3600,
-                "upstreamArtifacts": [
-                    {
-                        "paths": apks,
-                        "formats": [signing_format],
-                        "taskId": build_task_id,
-                        "taskType": "build"
-                    }
-                ]
+    def craft_ktlint_task(self):
+        return self._craft_gradle_clean_task(
+            name='ktlint',
+            description='Running ktlint over all modules',
+            gradle_task='ktlint',
+            treeherder={
+                'jobKind': 'test',
+                'machine': {
+                  'platform': 'lint',
+                },
+                'symbol': 'ktlint',
+                'tier': 1,
+            }
+        )
+
+    def craft_lint_task(self):
+        return self._craft_gradle_clean_task(
+            name='lint',
+            description='Running ktlint over all modules',
+            gradle_task='lint',
+            treeherder={
+                'jobKind': 'test',
+                'machine': {
+                  'platform': 'lint',
+                },
+                'symbol': 'lint',
+                'tier': 1,
             },
-            "provisionerId": "scriptworker-prov-v1",
-            "metadata": {
-                "name": name,
-                "description": description,
-                "owner": self.owner,
-                "source": self.source
+        )
+
+    def _craft_gradle_clean_task(
```

Comment:
Super nit: perhaps this can be named `_craft_clean_gradle_task`?
At the moment, when reading calls to this function, the name sounds like you're creating the "clean" task in gradle, and it's less obvious how the parameters affect the "clean" task. It sounds like the noun of "the clean task" is the important part, and you're configuring bits of it as the caller.

With the name `_craft_clean_gradle_task`, it sounds like we're invoking some gradle task, and "clean" is an adjective defining the task itself. So, the caller is defining a task to run, and the task will be "clean".

I hope that makes sense? It's suuuper minor, but it could subliminally increase readability?

Reply:
Thanks for pinpointing that. I agree with you. Function renamed.